dashboard/app/models/concerns/localizable.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# # Localizable
+#
+# A concern that provides automatic localization for specified model attributes.
+# Uses I18n with smart fallbacks to the original attribute values when translations
+# are missing or the locale is invalid.
+#
+# ## Setup
+# Include the concern and define which attributes should be localizable:
+#
+#   class CourseOffering < ApplicationRecord
+#     include Localizable
+#     self.localizable_attributes = :display_name, :description
+#   end
+#
+# ## I18n Structure
+# Translations are expected at: data.<model_plural>.<object_key>.<attribute>
+# Example: data.course_offerings.csp.display_name
+#
+# ## Usage
+# External usage - call on model instances
+#   course = CourseOffering.find_by(key: 'csp')
+#   course.localized_display_name                                    # Uses current I18n.locale
+#   course.localized_display_name('es')                              # Specify locale explicitly
+#   course.localized_display_name('es', key_override: 'custom-key')  # Override lookup key
+#   course.localized_display_name_with_options(locale: 'es', key: 'custom-key')
+#
+#   # Internal usage - within the model itself
+#   class CourseOffering < ApplicationRecord
+#     include Localizable
+#     self.localizable_attributes = :display_name, :description
+#
+#     def formatted_title
+#       "Course: #{localized_display_name}"
+#     end
+#
+#    def summarize_for_show
+#       {
+#         key: key,
+#         display_name: localized_display_name,
+#         header: formatted_title,
+#         description: localized_description,
+#       }
+#     end
+#   end
+#
+# ## Features
+# - Automatic method generation for configured attributes
+# - Smart caching with cache invalidation
+# - Graceful fallback to original values
+# - Error handling for invalid locales
+# - Validation of configured attributes
+
+module Localizable
+  extend ActiveSupport::Concern
+
+  included do
+    after_initialize :setup_localization_cache
+  end
+
+  class_methods do
+    def localizable_attributes
+      @localizable_attributes || []
+    end
+
+    # Class method to define localizable attributes
+    def localizable_attributes=(*attrs)
+      @localizable_attributes = attrs.flatten.map(&:to_sym)
+
+      # Generate localized_* methods for each configured attribute
+      @localizable_attributes.each do |attr|
+        define_method("localized_#{attr}") do |locale_code = I18n.locale, key_override: nil|
+          localize_property(attr, locale_code: locale_code, key_override: key_override)
+        end
+
+        # Also create a method that accepts options hash (for backward compatibility)
+        define_method("localized_#{attr}_with_options") do |options = {}|
+          locale_code = options[:locale] || options[:locale_code] || I18n.locale
+          key_override = options[:key]
+          localize_property(attr, locale_code: locale_code, key_override: key_override)
```

Comment:
Just an example of a new Ruby 3.1 feature: [Values in Hash literals and keyword arguments can be omitted](https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/#:~:text=Values%20in%20Hash%20literals%20and%20keyword%20arguments%20can%20be%20omitted)
```suggestion
          localize_property(attr, locale_code:, key_override:)
```